cmake/Buildretargeting.cmake

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2018  iCub Facility, Istituto Italiano di Tecnologia
+# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+
+include(YCMEPHelper)
+include(FindOrBuildPackage)
+
+find_or_build_package(YARP QUIET)
+find_or_build_package(ICUB QUIET)
+find_or_build_package(iDynTree QUIET)
+
+ycm_ep_helper(retargeting TYPE GIT
```

Comment:
The name with which a CMake package is identified in the superbuild is the CMake package name (i.e. the name with which you can find it using `find_package(<package_name> ..)` . For projects that do not install any config file, we tipically use the CMake's project name (the argument to the `project` call in CMake), as tipically  the project name and the package name in most CMake projects are coinciding. In this case, `regargeting` is the CMake project name of the repo: https://github.com/robotology/walking-teleoperation/blob/master/CMakeLists.txt#L8 . 
If we want to change it to align it with the repo, I suggest to modify it first in the `project` call in the repo.

Reply:
@traversaro Giulio and I decided to change the name of the project to `teleoperation` instead of `retargeting`. Do you agree with that? so, it has a similar name to the profile name!
If yes, I will change the names here, and in a walking-teleoperation PR, which I will make it upon your response.

Reply:
`teleoperation` is a bit generic and may be prone to collision, but if it is ok for you for me it is ok.

Reply:
@traversaro do you have any suggestions? What do you think about walking-teleoperation ?

Reply:
I definitely think `walking-teloperation` is definitely more specific.

Reply:
OK, nice. I will change the names on the PRs.